server/player_service.py

```diff
@@ -167,20 +176,37 @@ async def _fetch_player_legacy_rating(self, player, rating_types, conn):
                 )
                 continue
 
-
             player.ratings[rating_type] = (
                 row[table.format("mean")], row[table.format("deviation")]
             )
             player.game_count[rating_type] = row[table.format("numGames")]
 
-
     def remove_player(self, player: Player):
         if player.id in self._players:
             del self._players[player.id]
             metrics.players_online.set(len(self._players))
 
-    def get_permission_group(self, user_id: int) -> int:
-        return self.privileged_users.get(user_id, 0)
+    async def has_permission_role(self, player: Player, role_name: str) -> bool:
```

Comment:
Given that the `user_groups` are already pre-fetched and stored in the player object, maybe this could be quicker / work with only one join?

As far as I can see right now, all `user_groups` are loaded on `fetch_player_data`, but then only used to check if the player is a moderator (in the `mod` property), while checking any other permission uses this query?

Reply:
Yea I can if I also grab the group id's.